### ISSUE TITLE
About reward signal design in GRPO

### ISSUE THREAD

**Issue by lucasjinreal** [huggingface/open-r1#135]
hi, the default uses a Latex regex matching to get reward signal. If we want training on more general purpose, how to design this signal properly?

**Comment by ModeEric** [comment]
Hello,

Correct me if I'm wrong but I'm assuming your question is how to create a reasoning model that can output the answer in any format, not just the boxed format used by the Deepseek-R1 paper.

If that's the case, I would posit that the reason this is done is to get an unambiguous format and choice of answer for accurate reward modeling. If you want to create a reasoning model that doesn't adhere to such a format, you would need one of the following:

1. A method for extracting the answer from the "general" format you choose
2. Finetune the model using with GRPO for reasoning, and then follow up with SFT training to match your format but maintain the reasoning component.
3. While finetuning the GRPO model, have two distinct answer sections in series, one for the answer formatted you want, one for the "boxed" format, and train the language model to generate both, and then exclude the formatted one during inference.

Let me know if this answers your question!

**Comment by lucasjinreal** [comment]
Hi, the GRPO needs a reward signal right? Does current finetune on Math could work for general dataset? (Not just mathmatic related answers)

**Comment by ModeEric** [comment]
Yes, as far as I know all policy optimization methods (including GRPO) require some sort of reward signal. As a result, fine-tuning is clearly possible in the domain of math because we are able to check the accuracy of a solution, as well as the consistency of the responded format. Therefore, it certainly is possible to apply it to a general dataset as long as it is possible to extract a **meaningful reward signal** from that data. In the case you mentioned of value-alignment, you would need something like a ground truth label as to the appropriateness of a question, or a highly-accurate classifier that can do it automatically.

**Comment by lucasjinreal** [comment]
@ModeEric  Seems math problems can have a final answer or a close thinking process. But for some value alignment, or some truth knowledge refinement, any example dataset would be referenced to in GRPO?

**Comment by ModeEric** [comment]
You would need some way to score the "goodness" of a language response (accuracy, structure, style) for GRPO to work, regardless of your choice of dataset. To do this, you would need to sample a few responses, measure the "goodness" of such a response, and update with GRPO.
